x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -155,6 +155,7 @@ public LogicalPlan optimize(LogicalPlan verified) {
         if (failures.hasFailures()) {
             throw new VerificationException(failures);
         }
+        optimized.setOptimized();
```

Comment:
Doesn't the last rule in the optimizer, `SetAsOptimized()`, do that already?

Reply:
I'll double check it. I ran into something where it wasn't before.